_posts/2015-11-12-Using-OCR-To-Fix-A-Hilarious-Bug.markdown

```diff
@@ -0,0 +1,62 @@
+
+<!-- more -->
+
+Here's an example of a bug report where the thumbnail for [Marina Abramović](https://www.artsy.net/artist/marina-abramovic-1) became the text of her bio.
+
+![Bad Search](/images/2015-11-12-hilarious-bug/search.png)
+
+Here's one from our [iOS app](https://github.com/artsy/eigen) showing that thumbnails for related artists are set to their bios as well.
+
+![Bad Related Artists](/images/2015-11-12-hilarious-bug/eigen.png)
+
+Weird right? We eventually tracked down what was going on, and it's actually perfectly summarized in [this issue](https://github.com/blueimp/jQuery-File-Upload/pull/3356).
+
+We have an admin panel that allows some metadata about an artist to be edited. This includes their bio, as well as a place to upload a representative image as their 'cover thumbnail'.
+
+As the issue describes, we had some text input fields, as well as a file upload form using [Blueimp's jQuery File Upload](https://github.com/blueimp/jQuery-File-Upload). When you don't specify a `pasteZone` it defaults to the entire document. This means that a paste event anywhere on the page will trigger that event.
+
+Our [Art Genome Project Team](https://www.artsy.net/theartgenomeproject), which manages artist metadata, was using Microsoft Excel and Word to organize some of that data, including bios. When ready, a team member would copy and paste the bio into the bio input text field. This would also immediately fire the event for the image upload, which now automagically became an actual picture of the text of the bio. Our API and image processing pipeline would happily accept that, leading to the incredibly bizarre bug reports.
+
+My immediate fix was to specify and scope `pasteZone` (and similarly, `dropZone`) to the element the file upload widget was bound to, and try to figure out a way to programmaticaly detect these problematic images.
+
+Since the images were that of text, I decided to use OCR to remove artist thumbnails that it determined had 'too much text'. Now this may have unset valid covers from artists that use lots of text in their work, such as [Joseph Kosuth](https://www.artsy.net/artist/joseph-kosuth). In order to address that issue, prior to running my script I had added some functionality that will automatically set an artist thumbnail based on their iconic artworks, if one hadn't been explicitly set. This was nice because it meant that I could 'safely' unset thumbnails which my script flagged as having too much text, and the data would correct itself and become consistent once again, but with the actual problematic images gone. An artist with text-heavy works who had their thumbnail unset by the script would automatically get a new one set from their artworks.
```

Comment:
There's a little too much detail about internals here... maybe you can just say "...(this was safe, since we automatically fall back to the image of an artist's popular artwork)."